merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/engine/SimulationEngine.java

```diff
@@ -560,10 +561,44 @@ public static SimulationActivityExtract computeActivitySimulationResults(
         engine,
         startTime,
         elapsedTime,
-        computeTaskInfo(timeline, activityTopic, serializableTopics)
+        computeSpanInfo(timeline, activityTopic, serializableTopics)
     );
   }
 
+  public static HashMap<SpanId, ActivityDirectiveId> spanToActivityDirectiveId(
```

Comment:
This method should be private - in part because it's a helper method for computeResults, and in part because it requires the caller to provide a SpanInfo object, which is a private inner class of SimulationEngine.

```suggestion
  private static HashMap<SpanId, ActivityDirectiveId> spanToActivityDirectiveId(
```

Reply:
Got it made it private